### PR TITLE
catppuccin-gtk: unstable-2022-08-01 -> 0.2.7

### PR DESCRIPTION
**Pull request by kilianar**
###### Description of changes
https://github.com/catppuccin/gtk/releases/tag/v-0.2.7

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fufexan** [comment]
Doesn't the script install all the themes by default? (all colors, all variants). Why should we change it?

**Comment by kilianar** [comment]
You're right, the script does install all combinations of themes and colors by default. In my opinion it's useful to be able to install only the themes one needs/wants.

If you think it's unnecessary,  I'm happy to remove the feature.

**Comment by fufexan** [comment]
I've also made themes selectable at build time for the Orchis theme, but SuperSandro2000 (I think) said we should have them all available at once, just like the WhiteSur theme. So I'm just following the advice I was given.

**Comment by fufexan** [comment]
Looks good. Only thing I managed to overlook in the last PR was the mixup between Nord and Dracula variants.
```diff
-illegalMix = !(lib.elem "nord" tweaks) && !(lib.elem "dracula" tweaks);
+illegalMix = (lib.elem "nord" tweaks) && (lib.elem "dracula" tweaks);
```

I'd be grateful if you could also include this fix.

**Review by fufexan — APPROVED** [review]

**Comment by romildo** [comment]
Result of `nixpkgs-review pr 187245` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>catppuccin-gtk</li>
  </ul>
</details>

**Comment by romildo** [comment]
> You're right, the script does install all combinations of themes and colors by default. In my opinion it's useful to be able to install only the themes one needs/wants.
> 
> If you think it's unnecessary, I'm happy to remove the feature.

I also think it is useful to be able to install only the desired theme variants, although it can default to all. This is interesting when one wants to reduce the size of the package or reduce the list of options when selecting themes.

I have this feature with many theme and icon packages I maintain.

**Comment by fufexan** [comment]
You do have a point. Besides, the GitHub repo of the theme also has multiple variant artifacts which you download individually instead of getting everything.

I don't have the final say in this, but if any committer is fine with the configurable approach, I'm all for it.

**Review by romildo — APPROVED** [review]
LGTM. Tested on my system.

**Pull request merged**
